app/src/main/java/org/schulcloud/mobile/ui/homework/add/AddHomeworkPresenter.java

```diff
@@ -0,0 +1,138 @@
+package org.schulcloud.mobile.ui.homework.add;
+
+import org.schulcloud.mobile.data.DataManager;
+import org.schulcloud.mobile.data.local.PreferencesHelper;
+import org.schulcloud.mobile.data.model.Course;
+import org.schulcloud.mobile.data.model.CurrentUser;
+import org.schulcloud.mobile.data.model.Homework;
+import org.schulcloud.mobile.data.model.requestBodies.AddHomeworkRequest;
+import org.schulcloud.mobile.injection.ConfigPersistent;
+import org.schulcloud.mobile.ui.base.BasePresenter;
+import org.schulcloud.mobile.util.RxUtil;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import rx.Subscriber;
+import rx.Subscription;
+import rx.android.schedulers.AndroidSchedulers;
+import timber.log.Timber;
+
+@ConfigPersistent
+public class AddHomeworkPresenter extends BasePresenter<AddHomeworkMvpView>
+{
+    private PreferencesHelper mPreferencesHelper;
+
+    private Subscription mCoursesSubscription;
+    private Subscription mCurrentUserSubscription;
+
+    private List<Course> mCourses;
+
+    private DateFormat mDateFormat;
+
+    @Inject
+    public AddHomeworkPresenter(DataManager dataManager, PreferencesHelper preferencesHelper)
+    {
+        mDataManager = dataManager;
+        mPreferencesHelper = preferencesHelper;
+
+        mDateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm");
+    }
+
+    @Override
+    public void detachView()
+    {
+        super.detachView();
+        RxUtil.unsubscribe(mSubscription);
+        RxUtil.unsubscribe(mCoursesSubscription);
+        RxUtil.unsubscribe(mCurrentUserSubscription);
+    }
+
+    public void loadData()
+    {
+        checkViewAttached();
+
+        mCourses = new ArrayList<>();
+        mCourses.add(null);
```

Comment:
The user can select not to associate this homework with a course at all, just as in the web interface. For that case a dummy is inserted into both lists, and if the user selects "Keine Zuordnung", null is being sent to the server which results in no associated course.
https://github.com/schul-cloud/schulcloud-mobile-android/blob/01178ff91828e9b5d76f20ecb4b1cc6a0e79f605/app/src/main/java/org/schulcloud/mobile/ui/homework/add/AddHomeworkPresenter.java#L60-L63